endpoints/openrtb2/auction.go

```diff
@@ -1931,6 +1965,96 @@ func getAccountID(pub *openrtb2.Publisher) string {
 	return metrics.PublisherUnknown
 }
 
+// Returns the account from the request
+func getAccountIdFromRawRequest(hasStoredRequest bool, storedRequest json.RawMessage, originalRequest []byte) (string, bool, []error) {
+	var accountId string
+	var isAppReq bool
+	var err error
+
+	if hasStoredRequest {
+		accountId, isAppReq, err = searchAccountId(storedRequest)
+		if err != nil {
+			return "", isAppReq, []error{err}
+		}
+	} else {
+		accountId, isAppReq, err = searchAccountId(originalRequest)
+		if err != nil {
+			return "", isAppReq, []error{err}
+		}
+	}
+
+	if accountId == "" {
+		return metrics.PublisherUnknown, isAppReq, nil
+	}
+
+	return accountId, isAppReq, nil
+}
+
+func searchAccountId(request []byte) (accountId string, isAppReq bool, err error) {
+	var exists bool
+	// search in app.publisher.ext.prebid.parentAccount
+	accountId, exists, err = getAccountIdFromPublisherExtInRequest(request, "app")
+	if err != nil {
+		return "", true, err
+	}
+	if !exists {
+		// search in app.publisher.id
+		accountId, exists, err = getAccountIdAsPublisherIdInRequest(request, "app")
+		if err != nil {
+			return "", true, err
+		}
+	}
+	if exists {
+		isAppReq = true
+	}
+	if !exists {
+		// search in site.publisher.ext.prebid.parentAccount
+		accountId, exists, err = getAccountIdFromPublisherExtInRequest(request, "site")
+		if err != nil {
+			return "", false, err
+		}
+	}
+	if !exists {
+		// search in site.publisher.id
+		accountId, exists, err = getAccountIdAsPublisherIdInRequest(request, "site")
+		if err != nil {
+			return "", false, err
+		}
+	}
+
+	return accountId, isAppReq, err
+}
+
+func getAccountIdFromPublisherExtInRequest(request []byte, key string) (string, bool, error) {
+	accountId, dataType, _, err := jsonparser.Get(request, key, "publisher", "ext", openrtb_ext.PrebidExtKey, "parentAccount")
+	if dataType == jsonparser.NotExist {
+		return "", false, nil
+	}
+	if err != nil {
+		return "", false, err
+	}
+	if dataType != jsonparser.String {
+		return "", true, errors.New(key + ".publisher.ext.prebid.parentAccount must be a string")
+	}
+
+	return string(accountId), true, nil
+}
+
+func getAccountIdAsPublisherIdInRequest(request []byte, key string) (string, bool, error) {
+	accountId, dataType, _, err := jsonparser.Get(request, key, "publisher", "id")
+	if dataType == jsonparser.NotExist {
+		return "", false, nil
+	}
+	if err != nil {
+		return "", false, err
+	}
+	if dataType != jsonparser.String {
+		return "", true, errors.New(key + ".publisher.id must be a string")
+	}
+
+	return string(accountId), true, nil
+}
```

Comment:
Maybe it's better to simplify it to a single func?
Client code can look like this:
```
getAccountIdFromRequest(req.BidRequest.Ext, []string{"app", "key"})
getAccountIdFromRequest(req.BidRequest.Ext, []string{"site", "key"})
```

```suggestion
func getAccountIdFromRequest(request []byte, key []string) (string, bool, error) {
	accountId, dataType, _, err := jsonparser.Get(request, key...)
	if dataType == jsonparser.NotExist {
		return "", false, nil
	}
	if err != nil {
		return "", false, err
	}
	if dataType != jsonparser.String {
		return "", true, fmt.Errorf("%s must be a string", strings.Join(key, "."))
	}

	return string(accountId), true, nil
}
```

Reply:
done, but now its rather a multipurpose function, so I changed the name respectively